src/api/entities/DividendDistribution/index.ts

```diff
@@ -273,6 +274,64 @@ export class DividendDistribution extends CorporateAction {
     return paidStatuses.map((paid, index) => ({ ...participants[index], paid }));
   }
 
+  /**
+   * Retrieve an Identity that is entitled to dividends in this Distribution (participant),
+   *   the amount it is entitled to and whether it have been paid or not
+   *
+   * @note if the Distribution Checkpoint hasn't been created yet, the result will be null.
+   *   This is because the Distribution participant cannot be determined without a Checkpoint
+   */
+  public async getParticipant(args: {
+    identity: string | Identity;
+  }): Promise<DistributionParticipant | null> {
+    const {
+      targets: { identities: targetIdentities, treatment },
+      paymentDate,
+      perShare,
+      context,
+    } = this;
+
+    const checkpoint = await this.checkpoint();
+
+    if (checkpoint instanceof CheckpointSchedule) {
+      return null;
+    }
+
+    const isExclusion = treatment === TargetTreatment.Exclude;
+
+    const [did, balance] = await Promise.all([
+      getDid(args.identity, context),
+      checkpoint.balance(args),
+    ]);
+
+    const identity = new Identity({ did }, context);
+
+    const isTarget = !!remove([...targetIdentities], ({ did: targetDid }) => did === targetDid)
+      .length;
+
+    let participant: DistributionParticipant;
+
+    if (balance.gt(0) && xor(isTarget, isExclusion)) {
+      participant = {
+        identity,
+        amount: balance.multipliedBy(perShare),
+        paid: false,
+      };
+    } else {
+      return null;
+    }
+
+    // participant can't be paid before the payment date
+    if (paymentDate < new Date()) {
+      return participant;
+    }
+
+    const [paid] = await this.getParticipantStatuses([participant]);
```

Comment:
You should call directly to `context.polymeshApi.query.capitalDistribution.holderPaid` instead of this method. It chunks several calls and it does other stuff we don't need here.